.github/workflows/integration-tests.yaml

```diff
@@ -0,0 +1,285 @@
+---
+name: Integration Tests
+on:
+  pull_request:
+    paths:
+      - "action.yaml"
+      - ".github/workflows/integration-tests.yaml"
+
+env:
+  BRANCH: ${{ github.head_ref || (github.ref_type == 'branch' && github.ref_name) }}
+
+jobs:
+  setup:
+    name: Setup ${{ matrix.criteria.title }}
+    # These permissions are needed to:
+    # - Re-run a job from a workflow: https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#re-run-a-job-from-a-workflow-run
+    permissions:
+      actions: write
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # Order unfortunately matters here
+        criteria:
+          - title: Re-run Single Job
+            upload-on-run-attempts: "[1,2]"
+            fail-on-run-attempts: "[]"
+            re-run-mode: single-job
+          - title: Fallback
+            upload-on-run-attempts: "[1]"
+            fail-on-run-attempts: "[]"
+            re-run-mode: single-job
+          - title: Re-run Failed Jobs
+            upload-on-run-attempts: "[1,2]"
+            fail-on-run-attempts: "[1]"
+            re-run-mode: failed-jobs
+          - title: Re-run All Jobs
+            upload-on-run-attempts: "[1,2]"
+            fail-on-run-attempts: "[]"
+            re-run-mode: all-jobs
+    outputs:
+      json: ${{ steps.matrix-output.outputs.json }}
+    steps:
+      - name: Workflow Run Title
+        id: workflow-run-title
+        run: |
+          uuid="$(cat /proc/sys/kernel/random/uuid)"
+          workflow_title="Upload ${{ matrix.criteria.title }} ($uuid)"
+          echo "name=$workflow_title" | tee -a "$GITHUB_OUTPUT"
+      - name: Initiate upload workflow
+        uses: benc-uk/workflow-dispatch@e2e5e9a103e331dad343f381a29e654aea3cf8fc # v1.2.4
+        with:
+          workflow: upload.yaml
+          ref: ${{ env.BRANCH }}  # Workflow dispatch only works on branches or tags
+          inputs: |
+            {
+              "title": "${{ steps.workflow-run-title.outputs.name }}",
+              "upload-on-run-attempts": "${{ matrix.criteria.upload-on-run-attempts || '[]' }}",
+              "fail-on-run-attempts": "${{ matrix.criteria.fail-on-run-attempts || '[]' }}"
+            }
+      - run: sleep 5
+      # Utilizing a unique workflow run name to be able to identify the dispatched `run_id`
+      - name: Get workflow run
+        id: workflow-run
+        shell: bash
+        run: |
+          set -x
+          workflow_id="$(gh api -X GET --paginate "/repos/{owner}/{repo}/actions/workflows" --jq ".workflows[] | select(.path == \".github/workflows/${workflow_file:?}\").id")"
+          run="$(gh api -X GET --paginate "/repos/{owner}/{repo}/actions/runs" -f head_sha="${commit_sha:?}" --jq ".workflow_runs | map(select(.workflow_id == ${workflow_id:?} and .name == \"${workflow_run_name:?}\")) | .[0]")"
+          run_id="$(jq -er .id <<<"$run")"
+          echo "run-id=${run_id:?}" | tee -a "$GITHUB_OUTPUT"
```

Comment:
I should be able to get this feature into the `get-workflow-run` action eventually

Reply:
https://github.com/beacon-biosignals/get-workflow-run/issues/2